### PR TITLE
update(driver): add support for mlock2 syscall

### PR DESCRIPTION
**Pull request by alacuku**
```
Signed-off-by: Aldo Lacuku <aldo@lacuku.eu>
```

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area driver-kmod

/area driver-ebpf

/area libscap

/area libsinsp

> /area tests

> /area proposals

**What this PR does / why we need it**:
This PR adds support for the `mlock2` system call.

**Which issue(s) this PR fixes**:
falcosecurity/falco#1998

Fixes #

**Special notes for your reviewer**:
The `event_flag` for the sycall has been set to `EF_DROP_SIMPLE_CON`.

**Does this PR introduce a user-facing change?**:

```release-note
update(driver): add support for mlock2 syscall
```

### CODE REVIEW

File: driver/event_table.c [hunk 1]
```diff
@@ -374,6 +374,8 @@ const struct ppm_event_info g_event_info[PPM_EVENT_MAX] = {
 	/* PPME_GROUP_ADDED_X */{"groupadded", EC_PROCESS, EF_UNUSED, 0},
 	/* PPME_GROUP_DELETED_E */{"groupdeleted", EC_PROCESS, EF_MODIFIES_STATE, 3, {{"gid", PT_UINT32, PF_DEC}, {"name", PT_CHARBUF, PF_NA}, {"container_id", PT_CHARBUF, PF_NA} } },
 	/* PPME_GROUP_DELETED_X */{"groupdeleted", EC_PROCESS, EF_UNUSED, 0},
+	/* PPME_SYSCALL_MLOCK2_E */{"mlock2", EC_MEMORY, EF_DROP_SIMPLE_CONS, 3, {{"addr", PT_UINT64, PF_HEX}, {"len", PT_UINT64, PF_DEC}, {"flags", PT_UINT64, PF_HEX, mlock2_flags}}},
```
Comment (FedeDP) [inline driver/event_table.c:377]: I think the enter event should have 0 arguments,  given that you are using `sys_empty` filler for it :)

**Review by FedeDP — COMMENTED** [review]
Hi @alacuku ! Thanks for this PR!
I left a minor comment, but otherwise it LGTM!  
Note however that it won't be merged now because we are currently in the release process for Falco 0.32 :) So, bear with us, we will try our best to let it in after Falco 0.32 is out!

File: driver/event_table.c (hunk 1 above)
Comment (alacuku) [reply driver/event_table.c:377]: Thanks @FedeDP, I'll come back after Falco 0.32 is out!

File: driver/ppm_flag_helpers.h [hunk 2]
```diff
@@ -1591,6 +1591,16 @@ static __always_inline u32 mlockall_flags_to_scap(unsigned long flags)
 	return res;
 }
 
+static __always_inline u32 mlock2_flags_to_scap(unsigned flags)
```
Comment (Andreagit97) [inline driver/ppm_flag_helpers.h:1594]:
```suggestion
static __always_inline u32 mlock2_flags_to_scap(unsigned long flags)
```
I think that `unsigned long` here is safer :)

**Review by Andreagit97 — COMMENTED** [review]
LGTM! just a minor comment

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]
LGTM :+1: 

Thank you, Aldo!

**Pull request merged**